### ISSUE TITLE
Dimensions doesn't work as expected on initially entering split-view (iPad)

### ISSUE THREAD

**Issue by hblumberg** [facebook/react-native#28935]
## Description
It seems that when you initially enter split-view on an iPad, `Dimensions` does not update as expected. Specifically, the "change" event does not fire and `Dimensions.get('window').width` does not return the updated width.

After resizing the views, `Dimensions` goes back to updating as expected.

## React Native version:
(Note that the same behavior is observed on the latest version of React Native.)
```
React Native Environment Info:
    System:
      OS: macOS 10.15.2
      CPU: x64 Intel(R) Core(TM) i7-9750H CPU @ 2.60GHz
      Memory: 21.93 MB / 16.00 GB
      Shell: 3.2.57 - /bin/bash
    Binaries:
      Node: 10.20.1 - /usr/local/bin/node
      Yarn: 1.22.4 - /usr/local/bin/yarn
      npm: 6.14.4 - /usr/local/bin/npm
      Watchman: 4.9.0 - /usr/local/bin/watchman
    IDEs:
      Android Studio: 3.6 AI-192.7142.36.36.6308749
      Xcode: /undefined - /usr/bin/xcodebuild
    npmPackages:
      @sentry/react-native: ^1.1.0 => 1.1.0 
      react: 16.8.3 => 16.8.3 
      react-native: 0.59.10 => 0.59.10 
    npmGlobalPackages:
      react-native-cli: 2.0.1
```

## Steps To Reproduce
1. Open the app on an iPad.
2. See that `Dimensions.get('window').width` returns the expected width.
3. Enter split-view.
4. **See that the the dimensions change event isn't fired and that `Dimensions.get('window').width` returns the old width.**
5. Resize the app. See that the dimensions change event fires as expected and that `Dimensions.get('window').width` returns the expected width.

## Expected Results
1. Open the app on an iPad.
2. See that `Dimensions.get('window').width` returns the expected width.
3. Enter split-view.
4. **See that the the dimensions change event is fired and that `Dimensions.get('window').width` returns the new width.**
5. Resize the app. See that the dimensions change event fires as expected and that `Dimensions.get('window').width` returns the expected width.

## Snack and Screen Recording
https://snack.expo.io/@bryanjclark/rn-bug---dimensions
![Dimensions does not update on initial Split View](https://user-images.githubusercontent.com/8495791/82465785-a951d480-9a8d-11ea-8b60-f57663da54de.gif)

**Comment by bryanjclark** [comment]
(My naive hypothesis / suggestion for fixing this: what if Dimensions _always_ used the screen for dimensions, instead of sometimes window and sometimes screen?)

I'm building a workaround in our app for this, but at some point I'd like to come back and do a PR for this issue if nobody else has!

**Comment by bryanjclark** [comment]
Relates to https://github.com/facebook/react-native/issues/16152

**Comment by carbopilot** [comment]
It's also an issue on the iPad when moving an app from Slide Over to Split View mode and vice versa. The dimensions event listener in the the app does not get fired, yet the dimensions are different.

**Comment by hblumberg** [comment]
This issue has not yet been fixed.

**Comment by mym0404** [comment]
This issue hasn't fixed at `0.63.4`.
This is horrible.

**Comment by rdonnelly** [comment]
Can anyone provide an updated example project or Expo Snack the reproduces the issue?

One thing to note is that it is highly unlikely that things get back-ported to older versions, so making sure you're on the latest version of React Native and using up-to-date methods. I checked this out on my own and using 3 different methods for window size. Unless I'm missing something, I'm not seeing any issues on the latest Expo (40.0.0):

https://snack.expo.io/@rdonnelly/1cc697

Even the original original Snack seems to be working on Expo 40.0.0.

The window methods I tried are:
https://reactnative.dev/docs/usewindowdimensions
https://github.com/react-native-community/hooks#usedimensions
https://github.com/react-native-toolkit/react-native-responsive-dimensions
